sample-compose/data/src/main/java/co/nimblehq/sample/compose/data/repository/AppPreferencesRepositoryImpl.kt

```diff
@@ -0,0 +1,44 @@
+package co.nimblehq.sample.compose.data.repository
+
+import android.util.Log
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.*
+import co.nimblehq.sample.compose.domain.repository.AppPreferencesRepository
+import kotlinx.coroutines.flow.*
+import java.io.IOException
+import javax.inject.Inject
+
+class AppPreferencesRepositoryImpl @Inject constructor(
+    private val appPreferencesDataStore: DataStore<Preferences>
+) : AppPreferencesRepository {
+
+    private object PreferencesKeys {
+        val FIRST_TIME_LAUNCH = booleanPreferencesKey("FIRST_TIME_LAUNCH")
+    }
+
+    override fun isFirstTimeLaunchPreferences(): Flow<Boolean> = flow {
+        val firstLaunchPreferences = appPreferencesDataStore.data
```

Comment:
Naming it as `isFirstTimeLaunch` would be more consistent with `fun updateFirstTimeLaunchPreferences(isFirstTimeLaunch: Boolean)`

```suggestion
        val isFirstTimeLaunch = appPreferencesDataStore.data
```